python/python/Ice/Logger.py

```diff
@@ -1,16 +1,15 @@
 # Copyright (c) ZeroC, Inc.
 
-class Logger:
+from abc import ABC, abstractmethod
+
+class Logger(ABC):
     """
     The Ice message logger.
 
     Applications can provide their own logger by implementing this interface and installing it in a communicator.
     """
 
-    def __init__(self):
-        if type(self) is Logger:
-            raise RuntimeError("Ice.Logger is an abstract class")
-
+    @abstractmethod
     def _print(self, message):
```

Comment:
Not changed in this PR, but I think there is no need to scape `print` in a class method, maybe we should mark this as deprecated and add `print`.